src/components/workspace-utils.js

```diff
@@ -77,32 +100,104 @@ export const SnapshotInfo = ({ snapshotId, snapshotName }) => {
+            'aria-label': 'Edit description',
+            tooltip: 'Edit description'
+          }, [icon('edit')])
+        ]),
+        editingDescription ? h(Fragment, [
+          h(MarkdownEditor, {
+            options: {
+              autofocus: true,
+              placeholder: 'Enter a description',
+              renderingConfig: {
+                singleLineBreaks: false,
+                markedOptions: { sanitize: true, sanitizer: _.escape }
+              },
+              status: false
+            },
+            className: 'simplemde-container',
+            value: newDescription,
+            onChange: setNewDescription
+          }),
+          div({ style: { display: 'flex', justifyContent: 'flex-end', margin: '1rem' } }, [
+            h(ButtonSecondary, { onClick: () => setNewDescription(undefined) }, 'Cancel'),
+            h(ButtonPrimary, { style: { marginLeft: '1rem' }, onClick: save }, 'Save')
+          ])
+        ]) : h(MarkdownViewer, [description])
+      ]),
+      div({ style: { paddingLeft: '1rem' } }, [
+        div({ style: Style.dashboard.header }, ['Linked Data Repo Snapshot']),
+        h(SnapshotLabeledInfo, { title: 'Name:', text: sourceName }),
+        h(SnapshotLabeledInfo, { title: 'Creation Date:', text: Utils.makeCompleteDate(createdDate) }),
+        div({ style: { ...Style.elements.sectionHeader, marginBottom: '0.2rem' } }, ['Description:']),
+        div([sourceDescription]),
+        div({ style: Style.dashboard.header }, [`Source Data Repo Dataset${source.length > 1 ? 's' : ''}`]),
+        _.map(({ dataset: { id, name: datasetName, description: datasetDescription, createdDate: datasetCreatedDate } }) => {
+          return div({
+            key: id,
+            style: { marginBottom: '1rem' }
+          }, [
+            h(SnapshotLabeledInfo, { title: 'Name:', text: datasetName }),
+            h(SnapshotLabeledInfo, { title: 'Creation Date:', text: Utils.makeCompleteDate(datasetCreatedDate) }),
+            div({ style: { ...Style.elements.sectionHeader, marginBottom: '0.2rem' } }, ['Description:']),
+            div([datasetDescription])
+          ])
+        }, source)
+      ]),
+      editingName && h(Modal, {
+        onDismiss: () => {
+          setNewName(snapshotName)
+          setEditingName(false)
+        },
+        title: `Rename ${snapshotName}`,
+        okButton: h(ButtonPrimary, {
+          onClick: () => {
+            setEditingName(false)
+            save()
+          },
+          disabled: snapshotName === newName,
+          tooltip: snapshotName === newName && 'No change to save'
+        }, ['Rename'])
+      }, [
+        h(IdContainer, [id => h(Fragment, [
+          h(FormLabel, { htmlFor: id }, ['New snapshot name']),
+          h(TextInput, {
+            id,
+            autoFocus: true,
+            placeholder: 'Enter a name',
+            value: newName,
+            onChange: setNewName
+          })
```

Comment:
1. can you have this validate the input and disable the save button on an invalid name? Else it's far too easy for an end user to supply an illegal name and get an error on save. ("Name must be 1 to 63 alphanumeric characters or underscores, and cannot start with an underscore.")
2. errors on save appear not to be displayed - for me, they result in an infinite spinner with no feedback to the user.